sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java

```diff
@@ -47,22 +45,53 @@ public CommunicationTokenCredential(String token) {
      * Create with tokenRefreshOptions, which includes a token supplier and optional serialized JWT token.
      * If refresh proactively is true, callback function tokenRefresher will be called
      * ahead of the token expiry by the number of minutes specified by
-     * CallbackOffsetMinutes defaulted to ten minutes. To modify this default, call
-     * setCallbackOffsetMinutes after construction
+     * CallbackOffsetMinutes defaulted to ten minutes.
      *
      * @param tokenRefreshOptions implementation to supply fresh token when reqested
      */
     public CommunicationTokenCredential(CommunicationTokenRefreshOptions tokenRefreshOptions) {
-        Supplier<Mono<String>> tokenRefresher = tokenRefreshOptions.getTokenRefresher();
-        Objects.requireNonNull(tokenRefresher, "'tokenRefresher' cannot be null.");
-        refresher = tokenRefresher;
+        Supplier<String> tokenRefresher = tokenRefreshOptions.getTokenRefresherSync();
+        refresher = tokenRefresher != null
+                ? () -> Mono.fromSupplier(tokenRefresher)
+                : tokenRefreshOptions.getTokenRefresher();
+        Objects.requireNonNull(refresher, "'tokenRefresher' cannot be null.");
         if (tokenRefreshOptions.getInitialToken() != null) {
             setToken(tokenRefreshOptions.getInitialToken());
-            if (tokenRefreshOptions.isRefreshProactively()) {
-                OffsetDateTime nextFetchTime = accessToken.getExpiresAt().minusMinutes(DEFAULT_EXPIRING_OFFSET_MINUTES);
-                fetchingTask = new FetchingTask(this, nextFetchTime);
-            }
         }
+        if (tokenRefreshOptions.isRefreshProactively()) {
+            scheduleRefresher();
+        }
+    }
+
+    private void scheduleRefresher() {
+        OffsetDateTime nextFetchTime;
+        if (isTokenExpired(accessToken)) {
+            nextFetchTime = OffsetDateTime.now();
+        } else {
+            OffsetDateTime now = OffsetDateTime.now();
+            long tokenTtlMs = accessToken.getExpiresAt().toInstant().toEpochMilli() - now.toInstant().toEpochMilli();
+            nextFetchTime = isTokenExpiringSoon()
+                    ? now.plusNanos(TimeUnit.NANOSECONDS.convert(tokenTtlMs / 2, TimeUnit.MILLISECONDS))
```

Comment:
- `now.plusNanos` can be extracted one level up (less verbose + less code to maintain)
- let's extract `2` into a constant (similarly to [js](https://github.com/Azure/azure-sdk-for-js/blob/22acb7535a6c15f6003bbbc4d4a04dafd09ccbeb/sdk/communication/communication-common/src/autoRefreshTokenCredential.ts#L33))